Prevent stream sync from stopping when set, if playing

adriano-sudario opened the pull request:
If _AudioStreamSynchronized::set_sync_stream_ is called while playback is playing, it will stop - see [this issue](https://github.com/godotengine/godot/issues/101400)

I've described some of my experience with _AudioStreamSynchronized_ on [this bluesky thread](https://bsky.app/profile/did:plc:zximkmsmk54ssy6ekrxpoizl/post/3lcegipaalk2b), for those interested on discussing about it.
Also, [this PR](https://github.com/godotengine/godot/pull/100292) was set to draft and integrated here and some refactoring had to be made - as stated by [goatchurchprime](https://github.com/godotengine/godot/pull/100534#issuecomment-2585458774) - in order to work properly.

You can reproduce this behavior [here](https://github.com/adriano-sudario/godot_streams_issues_mrp), with detailed informations, running the _res://scene unexpected_behaviors/stream_sync_stops_when_set.tscn_

bruvzg committed:
[Export] Use project settings overrides with the target preset features instead of current platform features.

JoNax97 committed:
Turn 'Add/Revoke unique Node' into a checkbox

rarysson committed:
Add negative index to `Array.remove_at` and `Array.insert`

AThousandShips committed:
Add `String::replace_char(s)` methods for performance and convenience

bruvzg committed:
[TextEdit / LineEdit] Add support for OEM Alt codes input.

priahoud committed:
Add Options, Functions and Settings to convert Node-Names and Strings to kebab-case

- refactored and renamed String::_camelcase_to_underscore to String:_separate_compound_words
- refactored String::to_snake_case to work with the refactored String::_separate_compound_words
- created char_utils::is_hyphen to catch all hyphen variants in kebab-case conversion
- created String::to_kebab_case using the new String::_separate_compound_words
- created corresponding Documentation in String and StringName
- simplified both switch statements in EditorNode and ProjectDialog
- added new kebab-casing Option for Node Names in ProjectSettings
- added missing camelCase Options to Scene- and Node-Names in ProjectSettings
- simplified Mono RuntimeInterop Functions
- hooked up the ConnectionsDialog
- created additional Unit Tests

Repiteo committed:
Core: Add dedicated `BitField` template

Alex2782 committed:
Add configuration option to disable `Scroll Deadzone` on Android

HolonProduction committed:
Autocompletion: Don't add parenthesis if `Callable` is expected

KoBeWi committed:
Add create_id_for_path() to ResourceUID

Chaosus committed:
Re-organize UI in the shader editor

adriano-sudario committed:
Prevent stream sync from stopping when set if playing

adriano-sudario committed:
Update audio_stream_synchronized.cpp

adriano-sudario committed:
Update audio_stream_synchronized.cpp

adriano-sudario committed:
Update audio_stream_synchronized.cpp

Ivorforce committed:
Remove unnecessary `friend class` declarations of `CowData`.
Use default implementations for various containers.

Repiteo committed:
Core: Natively convert enum/BitField with Variant

goatchurchprime commented:
The AudioStreamSynchronized class came in 9 months ago from this PR https://github.com/godotengine/godot/pull/64488 from @reduz where there's a very extensive discussion about the features other than this one.

They've chosen to implement it with a fixed size array [Ref&lt;AudioStreamPlayback&gt; playback\[AudioStreamSynchronized::MAX_STREAMS\]](https://github.com/godotengine/godot/blob/master/modules/interactive_music/audio_stream_synchronized.h#L89C2-L89C73) and a `stream_count` value to state the working size of the array.

This code seems to have been made in a hurry, as shown by the fact that the [class parameters](https://github.com/godotengine/godot/blob/master/modules/interactive_music/audio_stream_synchronized.h#L91C1-L100C1) 

```C++
int play_order[AudioStreamSynchronized::MAX_STREAMS];

double stream_todo = 0.0;
int fade_index = -1;
double fade_volume = 1.0;
int play_index = 0;
double offset = 0.0;

int loop_count = 0;
```
have been accidentally pasted in from `class AudioStreamPlaybackPlaylist` and are not used!  So it's reasonable to expect shortcuts to have been taken, such as resetting all the streams when there has been a change.

Unfortunately there's also a bug: when you increase `stream_count` so it accesses more slots of the `playback` array, nothing gets reset even if there are valid streams in those slots.  This gives you a sneaky backdoor of setting `stream_count` to zero, changing a sync stream, and then setting `stream_count` back to its original value.

[These loops over `playbacks`, which generally has one element, are really confusing and made me think there were more bugs than there were.]

This fix looks good, and I've even tested it out.  

I have the following quibbles:

1. Should you call `playback_sync->playback[p_stream_index].unref()` before you set it to the new `p_stream` value?

2. If `set_stream_count()` increases the `stream_count`, should we initiate the playbacks and playing_pos of any non-null slots in the array of streams?  (Similarly, should we unref() the extra ones when `stream_count` decreases?)

3. How are looping tracks managed?  Perhaps we need a global playback position that you use to set the play positions of any looping tracks modulo their length?  This would give a handy way to unroll loops on tracks should you not want to watch their  `loop_count` value, especially if the AudioStreamSynchronized also maintained its own global_length.

adriano-sudario committed:
improvements on AudioStreamSynchronized class

adriano-sudario committed:
Update audio_stream_synchronized.cpp

HolonProduction committed:
Editor: Improve capsule gizmos

KoBeWi committed:
Add FoldableContainer

Co-authored-by: WhalesState <whalesstate@gmail.com>

cruglet committed:
Add middle-click removing for TabBars

Added a boolean option field to enable the firing of the "tab_close_pressed" signal when middle clicking on a tab.

van800 committed:
Fix .sln project generation logic for Rider to support all OS and all C++ toolchains

Co-authored-by: Andreia Gaita <shana@spoiledcat.net>
Co-authored-by: Rémi Verschelde <rverschelde@gmail.com>

ProgrammerOnCoffee committed:
Fix CodeEdit typos

adriano-sudario commented:
@adamscott can you take a look at this PR when you have time, please?

YYF233333 committed:
Dictionary::get_key_list use LocalVector instead of List.

YYF233333 committed:
Do not iterate Dictionary with get_key_at_index.

Mickeon committed:
Fix *even* more miscellaneous oddities around the class reference

Grublady committed:
Update inspector theme when changed

aaronfranke committed:
Add color space alpha notes to Color documentation

bruvzg committed:
AccessKit integration for macOS, Linux, and Windows.

bruvzg committed:
Initial editor accessibility.

YYF233333 committed:
Do not use Dictionary::keys() for Dictionary iteration.

bruvzg committed:
Base accessibility API.

Repiteo committed:
CI: Bump various pre-commit hooks

Rindbee committed:
Fix setting root inspector's `follow_focus` to `true` when `update_tree()` method ends

This may produce unexpected behavior in **recursive** calls.

van800 committed:
split Rider and Fleet as different external editors for Godot C#

Co-authored-by: Paul Joannon <437025+paulloz@users.noreply.github.com>
Co-authored-by: A Thousand Ships <96648715+AThousandShips@users.noreply.github.com>

goatchurchprime committed:
fix for stereo panning adding _calc_output_vol_stereo issue 103989

CreatedBySeb committed:
Prioritise static value type for dictionary properties

MewPurPur committed:
Optimize Color.html()

Aziroshin committed:
[CodeEdit] Fix folding for comments mixed with code region tags.

Co-authored-by: Kit Bishop <kitbdev@gmail.com>

syntaxerror247 committed:
Android: Hardware keyboard connection status callback

Adds a DisplayServer method to register hardware keyboard connection change callback.

syntaxerror247 committed:
Add an editor setting to enable/disable TouchActionsPanel

Adds an editor setting to enable/disable TouchActionsPanel.

Automatically hide the panel when hardware keyboard is connected.

Mickeon committed:
Fix miscellaneous oddities around the class reference (part 4)

Repiteo committed:
Merge pull request #105030 from Ivorforce/class-is-enabled-object

Move `GD_IS_CLASS_ENABLED` and respective include from `class_db.h` to `object.h`, as this is where it's needed.

Repiteo committed:
Merge pull request #101774 from Riteo/wayland-multiwindow-for-real-this-time

Wayland: Implement native sub-windows

Repiteo committed:
Merge pull request #105020 from YYF233333/opt_get_inheriters

Optimize `ClassDB::get_inheriters_from_class`

clayjohn committed:
Avoid using a global variable to store instance index in canvas items shader in RD renderer

lodetrick committed:
Fix Grabber Double Scaling when zoomed

lodetrick committed:
Consumes Scroll Events and fixes Mouse Warp

Mickeon committed:
Fix differences between PhysicsDirectSpaceState2D/3D docs

beicause committed:
PortableCompressedTexture: Support ASTC format and creating directly from compressed image

Mickeon committed:
Fix incorrect data format in RenderingDevice docs

Mickeon committed:
Fix differences between OpenXRInterface and XRHandTracker docs

Mickeon committed:
Fix typo in common note for packed arrays

Mickeon committed:
Fix typo in Window's `exclude_from_capture`

Repiteo committed:
Revert "CI: Remove "Free disk space on runner" job"

This reverts commit 6013fb1fb9bd9872f0ceac1827d77e31fbfccfbb.

HolonProduction committed:
LSP: Fix relative path handling for document links

HolonProduction committed:
LSP: Don't advertise support for workspace symbols

KoBeWi committed:
Add UID to file tooltip

lodetrick committed:
Fixes Editor Scales in OpenXR GUI

Commit:
add missing word to canvas layer docs

Delsin-Yu committed:
Avoid array allocation when signal have 0 arg

smix8 committed:
Combined TileMapLayer debug quadrant shapes to a surface

Create a combined mesh surface for all mesh faces and mesh lines in the TileMapLayer debug quadrant. Before it created a new mesh surface for each shape crashing into the mesh surface limit of 256 quickly.

Chubercik committed:
thorvg: Update to 0.15.12

Chubercik committed:
ufbx: Update to 0.18.0

bruvzg committed:
Add separate `minimize_disabled` and `maximize_disabled` window flags.

Ivorforce committed:
Fix caching of objects' class name pointer in `Object` instances.
Use `get_class_name` for `get_class` calls. Merge `get_parent_class_static` into `get_class_static`.

runbozhuang committed:
validate any form of running project

bruvzg committed:
Add custom sort to center tab bar in the editor and project manager.

jaydensipe committed:
Fix `Cannot get class` error when searching `TextFile`/`OtherFile`

bruvzg committed:
Fix EditorHelpBitTooltip in single window mode.

Ivorforce committed:
Handle the case where `waitpid` returns `errno` `EINTR`.
This case indicates that a debugger is attached, and `waitpid` should be called again.
Log errors when threads exit with `errno`.

Repiteo committed:
Merge pull request #105090 from smix8/tilemap_debug_surfaces

Combine `TileMapLayer` debug quadrant shapes to a surface

Repiteo committed:
Merge pull request #105095 from Mickeon/documentation-no-differences-XRHandTracker

Fix some differences between OpenXRInterface and XRHandTracker docs

Repiteo committed:
Merge pull request #105111 from bruvzg/nof_help

Fix `EditorHelpBitTooltip` in single window mode.

Repiteo committed:
Merge pull request #101003 from Repiteo/core/variant-native-enum

Core: Natively convert `enum`/`BitField` with `Variant`

Repiteo committed:
Merge pull request #104984 from KoBeWi/ETH

Fix wrong translation method in FileDialog

Repiteo committed:
Merge pull request #104991 from nikitalita/fix_inf_neg

`VariantWriter::write`: fix writing negative infinity when `p_compat` is true

Repiteo committed:
Merge pull request #104993 from lodetrick/editor-ruler-text-align

Fix Editor 2D Ruler Text Top-Align

Repiteo committed:
Merge pull request #104614 from kleonc/control_layout_mode_fallback_to_valid_value

Fallback `Control.layout_mode` to valid mode when child of non-container control

Repiteo committed:
Merge pull request #104756 from Rindbee/use-the-correct-root-inspector

Use the correct root inspector when making elements moved in an array visible

Repiteo committed:
Merge pull request #105006 from beicause/pctex-support-comp-image-astc

PortableCompressedTexture: Support ASTC format and creating directly from compressed image

Repiteo committed:
Merge pull request #105015 from syntaxerror247/improve-touchActionsPanel

Android Editor: Add an editor setting to enable/disable `TouchActionsPanel`

Repiteo committed:
Merge pull request #100287 from Chaosus/shader_ui_update

Re-organize UI in the shader editor

Repiteo committed:
Merge pull request #105024 from JulianHeuser/fix-tileset-custom-data

Fix editing custom data for tilesets

Repiteo committed:
Merge pull request #104897 from MewPurPur/optimize-color-from-string

Optimize Color.html()

Repiteo committed:
Merge pull request #104771 from Repiteo/ci/pre-commit-bump

CI: Bump various pre-commit hooks

Repiteo committed:
Merge pull request #105037 from clayjohn/RD-instance-index-crash

Avoid using a global variable to store instance index in canvas items shader in RD renderer

Repiteo committed:
Merge pull request #105039 from lodetrick/blend-tree-grabber

Fix EditorSpinSlider Grabber Scaling

Repiteo committed:
Merge pull request #104789 from jaydensipe/fix-textfile-class-error

Fix `Cannot get class` error when searching `TextFile`/`OtherFile` in FileSystem dock

Repiteo committed:
Merge pull request #105041 from lodetrick/editor-spin-slider-consume-input

Fix EditorSpinSlider Scroll Edit when inside a ScrollContainer

Repiteo committed:
Merge pull request #105053 from Mickeon/documentation-typo-RenderingDevice-DataFormat

Fix incorrect data format in RenderingDevice docs

Repiteo committed:
Merge pull request #105055 from Mickeon/documentation-typo-Windows-exclude-from-capture

Fix typo in Window's `exclude_from_capture`

Repiteo committed:
Merge pull request #105059 from HolonProduction/lsp-relative

LSP: Fix relative path handling for document links

Repiteo committed:
Merge pull request #105069 from KoBeWi/tUIDltip

Add UID to file tooltip

Repiteo committed:
Merge pull request #105061 from HolonProduction/lsp-workspace-symbol

LSP: Don't advertise support for workspace symbols

Repiteo committed:
Merge pull request #104432 from YYF233333/dict_iter2

Do not iterate `Dictionary` with `Dictionary::keys()`

Repiteo committed:
Merge pull request #105072 from lodetrick/action-map-editor-scale

Fix OpenXR Action Map GUI not scaling according to Editor Scale

Repiteo committed:
Merge pull request #105082 from Delsin-Yu/dotnet-signalawaiter-alloc-opt

[.Net] Avoid array allocation when signal have 0 arg

Calinou committed:
Unify Scripts Panel naming for the script and shader editors

Both use the same shortcut and action, so it makes sense to call it
a Files panel as opposed to a Scripts panel.

deralmas committed:
Wayland: Fix error spam for closed windows

I ported the new softclass macro too blindly. Apparently the various
window messages did not cast properly, skipping some important "deleted
window" checks.

bruvzg committed:
Add `uri_file_decode` to handle `+` in file names.

ProgrammerOnCoffee committed:
Reset `ProjectSettings` scroll when changing category

ProgrammerOnCoffee committed:
Edit misleading `Material::render_priority` description

timothyqiu committed:
Remove dead code in AnimationStateMachine editor

bruvzg committed:
Add bswap methods to the `PackedByteArray` bindings.

Ivorforce committed:
Fix `FileAccessCompressed` claiming one byte more than it actually has.
Add a unit test for fastlz compressed file access.

DeeJayLSP committed:
Add an actual name for Ogg Vorbis importer/visible name.

AThousandShips committed:
[Editor] Use separate `EditorFileDialog` for "Pack Project as ZIP"

This prevents state corruption for saving resources.

Repiteo committed:
Merge pull request #105099 from Ivorforce/object-classname-cache

Fix caching of objects' class name pointer in `Object` instances.

Repiteo committed:
Merge pull request #104333 from Mickeon/documentation-even-more-miscellaneous-oddities

Fix *even* more miscellaneous oddities around the class reference

Repiteo committed:
Merge pull request #102930 from Ivorforce/lambda-callable-overvalid

Optimize `GDScriptLambdaCallable` by skipping the unnecessary `ObjectDB` lookup for `script`.

Repiteo committed:
Merge pull request #105103 from runzh-crypto/validate_custom_directory_v2

Validate custom directory when running from editor

Repiteo committed:
Merge pull request #104339 from Grublady/inspector-bg-theme

Fix inspector theme not updating when theme changed

Repiteo committed:
Merge pull request #105106 from bruvzg/title_align

Add custom sort to center tab bar in the editor and project manager.

Repiteo committed:
Merge pull request #105107 from bruvzg/min-max-flag

Add separate `minimize_disabled` and `maximize_disabled` window flags.

Repiteo committed:
Merge pull request #104082 from YYF233333/dict_iter

Do not iterate `Dictionary` with `get_key_at_index`

Repiteo committed:
Merge pull request #105114 from Ivorforce/wait-for-pid-debugger

Handle the case where `waitpid` returns `errno` `EINTR`.

Repiteo committed:
Merge pull request #99099 from ydeltastar/align-to-orthogonal-view

Set correct position of node with `Align Transform with View` in orthographic view

Repiteo committed:
Merge pull request #76829 from bruvzg/ac_kit_direct

Implement screen reader support using AccessKit library.

Repiteo committed:
Merge pull request #105129 from Riteo/soft-serve

Wayland: Fix error spam for closed windows

Repiteo committed:
Merge pull request #105144 from Ivorforce/file-access-crompressed-size

Fix `FileAccessCompressed` claiming one byte more than it actually has.

Repiteo committed:
Merge pull request #103731 from GustJc/embed_pause_hotkeys

Make embed floating window respect `Always On Top` configuration

Repiteo committed:
Merge pull request #105152 from DeeJayLSP/ogg-name

Add an actual name for Ogg Vorbis importer visible name.

Repiteo committed:
Merge pull request #104264 from YYF233333/layout/gdvirtual

Optimize gdvirtual function layout

Repiteo committed:
Merge pull request #105154 from AThousandShips/fix_pack_zip

[Editor] Use separate `EditorFileDialog` for "Pack Project as ZIP"

Repiteo committed:
Merge pull request #102521 from HolonProduction/cancel-await

GDScript: Cancel suspended functions when reloading a script

adamscott committed:
Add `THREADS_ENABLED` check before RID thread tests

adamscott committed:
Move `THREADS_ENABLED` check after common imports

jaydensipe committed:
Add `Engine Version Update Mode` button to Project Manager `Quick Settings`

stuartcarnie committed:
Apple: Add pthread implementation of `Thread` class

This allows Apple platforms to override the default stack size of
a thread in the WorkerThreadPool, which is 512KiB by default.

This must be increased, as SPIRV-Cross, used by the Metal driver, can
use deeply nested stacks, as can debug builds.

Ivorforce committed:
Change `get_class_static` to return `StringName`.
Use that static `get_class_static` for `_get_class_namev` to avoid duplication.

ProgrammerOnCoffee committed:
Fix `UDPServer::poll` typos

bruvzg committed:
Fix initial project manager size and scales < 1.

bruvzg committed:
Remove error print in the Animation getter.

clayjohn committed:
Detect possible usage of SSS and screen textures at load time to allocate needed resources at load time.

This allows us to avoid a class of pipeline compilation stutters and
stutters from allocating screen textures.

clayjohn committed:
Detect possible usage of SSS and screen textures at load time to allocate needed resources at load time.

This allows us to avoid a class of pipeline compilation stutters and
stutters from allocating screen textures.

bruvzg committed:
Remove misleading and incorrect notes about endianness. Fix FileAccess and StreamPeer not doing what name suggests.

bruvzg committed:
Fix Thread crash when using invalid callback signature. And invalid message formatting.

lodetrick committed:
Consolidate Shader Menu in plugin

Repiteo committed:
Merge pull request #104853 from goatchurchprime/gtch/stereofix

Fix AudioStreamPlayer3D stereo panning issue

Repiteo committed:
Merge pull request #105135 from timothyqiu/delete-window

Remove dead code in AnimationStateMachine editor

Repiteo committed:
Merge pull request #105124 from Calinou/script-shader-editor-unify-toggle-files-panel

Unify Scripts Panel naming for the script and shader editors

Repiteo committed:
Merge pull request #105157 from adamscott/test_rid_add_threads_enabled

Add `THREADS_ENABLED` check before RID thread tests

Repiteo committed:
Merge pull request #102346 from KoBeWi/folding_whalegacy

Add `FoldableContainer`

Repiteo committed:
Merge pull request #105159 from adamscott/define_my_int

Move `THREADS_ENABLED` check after common imports

Repiteo committed:
Merge pull request #105166 from Ivorforce/get-class-static-stringname

Change `get_class_static` to return `StringName`.

Repiteo committed:
Merge pull request #105170 from ProgrammerOnCoffee/fix-udpserver-poll-typos

Fix typos in `UDPServer::poll` documentation

Repiteo committed:
Merge pull request #83027 from rarysson/array-negative-indexes

Add negative index to `Array.remove_at` and `Array.insert`

Repiteo committed:
Merge pull request #105173 from bruvzg/sub_1_ui_scale

Fix initial project manager size and scales < 1.

Repiteo committed:
Merge pull request #105178 from bruvzg/endianess_notes

Remove misleading and incorrect notes about endianness. Fix FileAccess and StreamPeer not doing what name suggests.

Repiteo committed:
Merge pull request #105174 from bruvzg/anim_err

Remove error print in the Animation getter.

Repiteo committed:
Merge pull request #105180 from bruvzg/thread_crash

Fix Thread crash when using invalid callback signature. And invalid message formatting.

Repiteo committed:
Merge pull request #104410 from aaronfranke/color-space-doc

Add color space alpha notes to Color documentation

Repiteo committed:
Merge pull request #104939 from HolonProduction/json-rpc-notification-return-fix

JSONRPC: Fix notification return behavior

Repiteo committed:
Merge pull request #103405 from van800/shakhov/macosx

Fix `.sln` project generation logic for Rider to support all OS and all C++ toolchains

Repiteo committed:
Merge pull request #100333 from YYF233333/reduce_list

Use `LocalVector` instead of `List` as arg of `Dictionary::get_key_list`

Repiteo committed:
Merge pull request #101745 from Rindbee/remove-is_top_level_control

Remove confusing `Control::is_top_level_control()`

Repiteo committed:
Merge pull request #71542 from bruvzg/export_project_settings

[Export] Use project settings overrides with the target preset features instead of current platform features.

bruvzg committed:
Remove focus draw for editor inspector elements.

Monstrofil committed:
Fix incorrect mipmap size calculation for uncompressed DDS textures

Previously, the DDS texture loader incorrectly calculated mipmap dimensions
for uncompressed textures when the original dimensions
were non-power-of-two.

For example, given a 120×120 texture, the expected mipmap chain should be:
    mipmap=0, size=120×120
    mipmap=1, size=60×60
    mipmap=2, size=30×30
    mipmap=3, size=15×15
    mipmap=4, size=7×7
    mipmap=5, size=3×3
    mipmap=6, size=1×1

But the loader was producing:
    mipmap=0, size=120×120
    mipmap=1, size=60×60
    mipmap=2, size=30×30
    mipmap=3, size=15×15
    mipmap=4, size=8×8
    mipmap=5, size=4×4
    mipmap=6, size=2×2

This commit corrects the logic to properly compute width and height independently at each mip level.

Fixes issue #105136.

aaronp64 committed:
Use validated key in Dictionary::has

Updated Dictionary::has to use validated key when checking variant_map, to handle cases when passed in key is converted to Dictionary's key type.

nikitalita committed:
add GDSOFTCLASS to FileAccess and DirAccess derived classes

bruvzg committed:
Remove "dummy" renderer from the editor dropdown.

smix8 committed:
Replace NavigationServer3D Point struct with Vector3

Replaces NavigationServer3D Point struct with Vector3.

smix8 committed:
Replace NavigationServer2D Point struct with Vector2

Replaces NavigationServer2D Point struct with Vector2.

Repiteo committed:
Merge pull request #104451 from WinnerWind/fix_resized_logic_for_control

Fix excessively calling resized and floating point errors in Control.

Repiteo committed:
Merge pull request #105096 from Chubercik/ufbx-0.18.0

ufbx: Update to 0.18.0

Repiteo committed:
Merge pull request #103306 from Calinou/macos-intel-vulkan-disable-texture-array-reflections

Disable texture array reflections on Intel GPUs on macOS due to driver bugs

Repiteo committed:
Merge pull request #101011 from aaronfranke/limit-max-contacts-reported

Place a hard limit on the `max_contacts_reported` property in 2D/3D physics

Repiteo committed:
Merge pull request #102292 from HolonProduction/gizmo-capsule

Editor: Improve capsule gizmos

Repiteo committed:
Merge pull request #104213 from ryevdokimov/shortcuts-gridmap-mouse-captrued

Fix `GridMap` shortcuts being triggered during freelook (mouse captured)

Repiteo committed:
Merge pull request #104357 from FilipeAlexCosta/control-set-position

Fix `Control.set_position` resizes offsets/anchors

Repiteo committed:
Merge pull request #105130 from bruvzg/uri_fix_plus

Add uri_file_decode to handle + in file names.

Repiteo committed:
Merge pull request #105126 from ProgrammerOnCoffee/fix-codeedit-typos

Fix CodeEdit typos

Repiteo committed:
Merge pull request #105005 from beicause/pctex-fix-basis-crash

PortableCompressedTexture: Fix BasisU crash and wrong format

Repiteo committed:
Merge pull request #104632 from ibrahn/popup-menu-mouseover-calc

Fix and simplify hit calculation in `PopupMenu::_get_mouse_over`.

Repiteo committed:
Merge pull request #94143 from Mickeon/documentation-object-connect-to-signal-too-long-didn't-read

Move and simplify Object's `connect` description slightly

Repiteo committed:
Merge pull request #104780 from aaronfranke/gltf-fix-invalid-byte-stride

GLTF: Fix importing files with invalid buffer view byte strides

Repiteo committed:
Merge pull request #104804 from etherealxx/fix-advanced-import-inspector-missing-tooltip

Fix missing tooltips on "Advanced Import Settings for Scene" window

Repiteo committed:
Merge pull request #105193 from Monstrofil/fix-dds-mipmap-size-uncompressed-dds

Fix incorrect mipmap size calculation for uncompressed DDS textures

Repiteo committed:
Merge pull request #105205 from aaronp64/dictionary_has_validated_key

Use validated key in `Dictionary::has`

Repiteo committed:
Merge pull request #105075 from heyitszackb/add-clarity-to-canvas-layer-docs

Add missing word to canvas layer docs

Repiteo committed:
Merge pull request #103805 from rsubtil/fix-recovery_lock_on_import

Fix recovery mode lock file not being cleared on import/export

Repiteo committed:
Merge pull request #104830 from chocola-mint/fix-#104822

Add null check for SpriteFramesEditor's SpriteFrames

bruvzg committed:
Change `DisplayServerMacOS` from `GDCLASS` to `GDSOFTCLASS`. Add `GDSOFTCLASS` to other display servers.

bruvzg committed:
Fix visual shader editor popups not using editor scale.

shana committed:
Fix issue where vsproj=yes vsproj_gen_only=no sometimes fails to build

Assigning `env["CCFLAGS"]` directly to a variable causes any changes to that
variable to also be done to the original env object. This means CCFLAGS would
be modified during generation of the VS project with additional flags and
other options set by platforms.

This would normally not be noticed when just generating the project, but when
generating and building at the same time with a compiler that doesn't support
VS-style flags, this flag leakage can cause unexpected build failures.

adamscott committed:
Add the equivalent of `-Werror` for `wasm-ld`

Repiteo committed:
Merge pull request #102786 from Giganzo/top-toolbar-menu-button-styles

Fix MenuButton style in editor top menu bar

Repiteo committed:
Merge pull request #102878 from Giganzo/tree-jiggle

Fix Tree buttons jiggle on horizontal scrolling

Repiteo committed:
Merge pull request #105226 from bruvzg/vs_popup

Fix visual shader editor popups not using editor scale.

Repiteo committed:
Merge pull request #105164 from stuartcarnie/apple_pthread

Apple: Add pthread implementation of `Thread` class

Repiteo committed:
Merge pull request #105133 from ProgrammerOnCoffee/edit-material-renderpriority-doc

Fix misleading `Material::render_priority` description

Repiteo committed:
Merge pull request #105007 from Aziroshin/dev/aziroshin/comment-after-region-not-folding-102382

[CodeEdit] Fix folding for comments mixed with code region tags.

Repiteo committed:
Merge pull request #103802 from Rindbee/set-unsupported-to-true-on-error-in-FreeDesktopScreenSaver

Set `unsupported` to true on error in `FreeDesktopScreenSaver`

Repiteo committed:
Merge pull request #103791 from BrotherShort/scenetree_item_text_display

Fix SceneTree Item text display bug in Right-to-Left

Repiteo committed:
Merge pull request #104785 from Rindbee/fix-root-inspector-follow_focus-set-to-true

Fix setting root inspector's `follow_focus` to `true` when `update_tree()` method ends

Repiteo committed:
Merge pull request #92475 from AThousandShips/string_replace_char

Add `String::replace_char(s)` methods for performance and convenience

Repiteo committed:
Merge pull request #103798 from Rudolph-B/new-97712

Switch occlusion culling to be based on depth instead of Euclidean distance

Repiteo committed:
Merge pull request #105175 from clayjohn/RD-pipeline-sss-roughness

Detect more pipeline settings at load time to avoid pipeline stutters

Repiteo committed:
Merge pull request #100086 from erodozer/multipart-ext-import

Support multi dot extensions in import plugins

Repiteo committed:
Merge pull request #105051 from Mickeon/documentation-no-differences-PhysicsDirectSpaceState

Fix differences between PhysicsDirectSpaceState2D/3D docs

Repiteo committed:
Merge pull request #104378 from DarkiStar/master

Documentation: Fix `hex_decode()` example in `String` and `StringName`

Repiteo committed:
Merge pull request #97852 from Jan200101/PR/remove-PrefersNonDefaultGPU

Remove PrefersNonDefaultGPU from linux desktop file

Repiteo committed:
Merge pull request #103198 from Naputt1/fix-tween.kill()-isValid()-return-false

Fix `Tween.is_valid()` returning `true` after calling `kill()`

Repiteo committed:
Revert early version of PR #105175

This reverts commit 6225e25f4adad3cacbef32cfa8607767b5a96269.

Repiteo committed:
Merge pull request #105175 from clayjohn/RD-pipeline-sss-roughness

Detect more pipeline settings at load time to avoid pipeline stutters

Repiteo committed:
Core: Use `Math` namespace for constants

Nolkaloid committed:
Add gdscript tests for callv with const arrays

Repiteo committed:
Merge pull request #94883 from priahoud/add-kebab-case-options-and-functions

Fix ProjectSettings name casing options and missing String methods

Repiteo committed:
Merge pull request #93636 from Nolkaloid/fix-const-callv

GDScript: Add tests for calling with wrong arguments in Callable.callv() when passing a readonly (const) Array

Repiteo committed:
Merge pull request #105162 from jaydensipe/add-update-check-button-to-quick-settings

Add `Engine Version Update Mode` button to Project Manager `Quick Settings`

Repiteo committed:
Merge pull request #105225 from bruvzg/ds_gdsoft

Change `DisplayServerMacOS` from `GDCLASS` to `GDSOFTCLASS`. Add `GDSOFTCLASS` to other display servers.

Repiteo committed:
Merge pull request #87756 from ryevdokimov/alt-transform

Allow all tool modes to select

Repiteo committed:
Merge pull request #105238 from shana/shana/vsgen-fix

Fix issue where vsproj=yes vsproj_gen_only=no sometimes fails to build

Repiteo committed:
Merge pull request #103024 from cruglet/master

TabBar: Add boolean toggle for middle-click to fire `tab_close_pressed` signal.

Repiteo committed:
Merge pull request #96375 from HolonProduction/completion-callable-braces

Autocompletion: Don't add parenthesis if `Callable` is expected

Repiteo committed:
Merge pull request #93466 from bruvzg/alt_oem

[TextEdit / LineEdit] Add support for OEM Alt codes input.

Repiteo committed:
Merge pull request #105145 from bruvzg/swap_pba

Add bswap methods to the `PackedByteArray` bindings.

bruvzg committed:
Fix graph node resizing.

nathanhoad committed:
Fix issue with playing a custom scene from a UID

Co-Authored-By: Tomasz Chabora <kobewi4e@gmail.com>

Repiteo committed:
Merge pull request #105216 from bruvzg/no_dummy_rs

Remove "dummy" renderer from the editor dropdown.

Repiteo committed:
Merge pull request #105249 from Repiteo/core/math-defs-namespace

Core: Use `Math` namespace for constants

Repiteo committed:
Merge pull request #105252 from smix8/nav2d_point

Replace NavigationServer2D `Point` struct with `Vector2`

Repiteo committed:
Merge pull request #105093 from Chubercik/thorvg-0.15.12

thorvg: Update to 0.15.12

Repiteo committed:
Merge pull request #105253 from smix8/nav3d_point

Replace NavigationServer3D `Point` struct with `Vector3`

Repiteo committed:
Merge pull request #82216 from JoNax97/unique-node-checkbox

Make "access as/revoke unique name" a checkbox

Repiteo committed:
Merge pull request #104772 from CreatedBySeb/fix-null-dict-values

Handle null values for object types, as seen in typed dictionaries

Repiteo committed:
Merge pull request #105210 from nikitalita/gdsoftclass-fa-da

Add GDSOFTCLASS to FileAccess and DirAccess derived classes

Repiteo committed:
Merge pull request #96139 from Alex2782/reverting_pr_84331

Add configuration option to disable `Scroll Deadzone` on Android

Repiteo committed:
Merge pull request #105132 from ProgrammerOnCoffee/fix-projectsettings-scroll

Reset `ProjectSettings` scroll when changing category

Repiteo committed:
Merge pull request #105265 from bruvzg/gr_node_resize

Fix graph node resizing.

Repiteo committed:
Merge pull request #105197 from bruvzg/inspector_no_focus_draw

Remove focus draw for editor inspector elements.

Repiteo committed:
Merge pull request #105073 from Mickeon/documentation-miscellaneous-oddities-part-4

Fix miscellaneous oddities around the class reference (part 4)

Repiteo committed:
Merge pull request #105242 from adamscott/add-wasm-ld-wall

[Web] Add the equivalent of `-Werror` for `wasm-ld`

Repiteo committed:
Merge pull request #99543 from KoBeWi/using_rng_to_destroy_rng

Add `create_id_for_path()` to ResourceUID

Repiteo committed:
Merge pull request #105277 from nathanhoad/fix-play-custom-scene-with-uid

Fix issue with playing a custom scene from a UID

Repiteo committed:
Merge pull request #104828 from van800/rider_fleet

C#: Separate Rider and Fleet options as external editors

Repiteo committed:
Merge pull request #105183 from lodetrick/shader-ui-shuffle

Make shader editor menu position consistent with script editor

zaevi committed:
[.NET] Fix `string.PathJoin` to be consistent with core

Repiteo committed:
Rendering: Fix `Math` constant conversion

Repiteo committed:
Merge pull request #105281 from zaevi/fix_csharp_string_pathjoin

[.NET] Fix `string.PathJoin` to be consistent with core

Repiteo committed:
Merge pull request #105286 from Repiteo/core/math-const-shader-fix

Rendering: Fix `Math` constant conversion

Repiteo committed:
Merge pull request #100650 from Ivorforce/cowdata-enhance-loneliness

Remove unnecessary `friend class` declarations of `CowData`.

Repiteo committed:
Merge pull request #95916 from Repiteo/core/bit-field

Core: Add dedicated `BitField` template

Repiteo committed:
Merge pull request #105054 from Mickeon/documentation-typo-packed-arrays

Fix typo in common note for packed arrays

adriano-sudario committed:
Create test_audio_stream_synchronized.h

adriano-sudario committed:
added tests wip

adriano-sudario committed:
Merge branch 'prevent-stream-sync-from-stopping-when-set' of https://github.com/adriano-sudario/godot into prevent-stream-sync-from-stopping-when-set

Calinou commented:
- Superseded by https://github.com/godotengine/godot/pull/106608.

Calinou commented:
- Superseded by https://github.com/godotengine/godot/pull/106608.
